src/Platform/PlatformCommandTrait.php

```diff
@@ -52,14 +52,14 @@ protected function initialize(InputInterface $input, OutputInterface $output) {
   public function addPlatform(string $alias, PlatformInterface $platform): void {
     $options = static::getExpectedPlatformOptions();
     foreach ($options as $name => $expectation) {
-      if (empty($this->platforms[$name])) {
+      if (empty($this->platforms[$name]) && empty($this->platformAliases[$alias])) {
         $event = new AddPlatformToCommandEvent($expectation, $platform, $alias);
         $this->dispatcher->dispatch(CommonConsoleEvents::ADD_PLATFORM_TO_COMMAND, $event);
         if (!$event->platformMatchesExpectation()) {
           throw new \Exception(sprintf("Invalid Platform value. Expected a platform of type '%s'. Type of '%s' given.", $expectation, $platform::getPlatformId()));
         }
         $this->platforms[$name] = $platform;
-        $this->platformAliases[$name] = $alias;
+        $this->platformAliases[$alias] = $name;
```

Comment:
why was this swapped? The idea is to be able to $this->getPlatform($name) and $this->getPlatformAlias($name) and get platform and alias for the same internal name. If we swap alias for name, we have to search array values instead of using the map's key.

Reply:
I understand, however I need to make sure that the platform has not been added to the list already. Either way I'm going to need to search in the array if we don't use the alias as the key. On the other hand there is no such method as getPlatformAlias(). Would you like me to add it to this trait? The latter one also contributed to my reasoning, because we didn't use this member variable except in PlatformSites.php where it can be swapped by calling getAlias() on platform directly.